app/database/group_item_store_compute_all_access.go

```diff
@@ -39,35 +53,7 @@ func (s *GroupItemStore) computeAllAccess() {
 		FROM items_items
 		JOIN groups_items AS parents
 			ON parents.item_id = items_items.parent_item_id
-		JOIN groups_items_propagate AS parents_propagate
-			ON parents.id = parents_propagate.id AND parents_propagate.propagate_access = 'children'`
-	stmtInsertMissingChildren, err = s.db.CommonDB().Prepare(queryInsertMissingChildren)
-	mustNotBeError(err)
-	defer func() { mustNotBeError(stmtInsertMissingChildren.Close()) }()
-
-	// inserting missing (or set propagate_access='self' to existing) groups_items_propagate
-	// for groups_items having propagate_access='self'
-	const queryInsertMissingPropagate = `
-		INSERT INTO groups_items_propagate (id, propagate_access)
-		SELECT
-			groups_items.id,
-			'self' as propagate_access
-		FROM groups_items
-		WHERE propagate_access='self'
-		ON DUPLICATE KEY UPDATE propagate_access='self'`
-	stmtInsertMissingPropagate, err = s.db.CommonDB().Prepare(queryInsertMissingPropagate)
-	mustNotBeError(err)
-	defer func() { mustNotBeError(stmtInsertMissingPropagate.Close()) }()
-
-	// Set groups_items as set up for propagation
-	// (switch groups_items.propagate_access from 'self' to 'done')
-	const queryUpdatePropagateAccess = `
-		UPDATE groups_items
-		SET propagate_access='done'
-		WHERE propagate_access='self'`
-	stmtUpdatePropagateAccess, err = s.db.CommonDB().Prepare(queryUpdatePropagateAccess)
-	mustNotBeError(err)
-	defer func() { mustNotBeError(stmtUpdatePropagateAccess.Close()) }()
+		JOIN parents_propagate ON parents_propagate.id = parents.id`
 
 	// mark as 'done' groups_items_propagate that shouldn't propagate (having items.custom_chapter=1)
 	const queryMarkDoNotPropagate = `
```

Comment:
> Wouldn't be faster (maybe slower actually) to join with groups_items_propagate and only apply this to propagate_access!='done' ?

Probably it would be worse: with the proposed modification MySQL uses an internal temporary table:
```mysql> explain INSERT INTO groups_items_propagate (id, propagate_access) SELECT groups_items.id AS id, 'done' as propagate_access FROM groups_items JOIN items ON groups_items.item_id = items.id AND items.custom_chapter LEFT JOIN groups_items_propagate AS prop ON prop.id = groups_items.id AND prop.propagate_access = 'done' WHERE prop.id IS NULL ON DUPLICATE KEY UPDATE propagate_access='done';
+----+-------------+------------------------+------------+--------+--------------------------+------------+---------+--------------------------------------+------+----------+------------------------------+
| id | select_type | table                  | partitions | type   | possible_keys            | key        | key_len | ref                                  | rows | filtered | Extra                        |
+----+-------------+------------------------+------------+--------+--------------------------+------------+---------+--------------------------------------+------+----------+------------------------------+
|  1 | INSERT      | groups_items_propagate | NULL       | ALL    | NULL                     | NULL       | NULL    | NULL                                 | NULL |     NULL | NULL                         |
|  1 | SIMPLE      | groups_items           | NULL       | index  | item_id,itemtem_id       | itemtem_id | 8       | NULL                                 |    1 |   100.00 | Using index; Using temporary |
|  1 | SIMPLE      | items                  | NULL       | eq_ref | PRIMARY                  | PRIMARY    | 8       | algorea_db_test.groups_items.item_id |    1 |   100.00 | Using where                  |
|  1 | SIMPLE      | prop                   | NULL       | eq_ref | PRIMARY,propagate_access | PRIMARY    | 8       | algorea_db_test.groups_items.id      |    1 |   100.00 | Using where; Not exists      |
+----+-------------+------------------------+------------+--------+--------------------------+------------+---------+--------------------------------------+------+----------+------------------------------+```

Reply:
> For the next next, queryMarkExistingChildren, it is over-complicated by the fact they want a (completely useless) ID for their entry 😕. Meaning also much more complicated to understand than what it is really.

They are not completely useless

Reply:
The key should be `group_id,item_id`. Here it is not clear that it is actually a key for the other table :-/, but it could be worst, with an ID + groups_items_id, true.

Reply:
But it forces going through the groups_items while it may not be needed.